clusterloader2/pkg/measurement/common/probes/probes.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probes
+
+import (
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/pkg/flags"
+	"math"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/config"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/prometheus"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	name            = "Probes"
+	probesNamespace = "probes"
+
+	manifestGlob = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/probes/manifests/*.yaml"
+
+	checkProbesReadyInterval = 30 * time.Second
+	checkProbesReadyTimeout  = 5 * time.Minute
+
+	currentProbesMetricsVersion = "v1"
+)
+
+var probesEnabled bool
+
+func init() {
+	flags.BoolEnvVar(&probesEnabled, "enable-probes", "ENABLE_PROBES", false, "Whether to run the probes in the clusterloader2 tests.")
```

Comment:
Why is this needed?
You need to explicitly perform actions on this measurement, so I don't think this additional guard really provides value.

Reply:
OK - I got your point. Kubemark is definitely a valid example.

I think it's a different thing that chaos-monkey - it's a measurement that is needed for SLO measuring. Let me think about it a bit more.

Reply:
I was thinking a bit about it and my current thinking about it is:

Kubemark is quite specific setup - some things just doesn't make sense. I tend to think that at some point we may be forced to fork our tests for kubemark, but obviously I would like to delay it by as much as possible.

This is the same case for measurements - some just may not make sense in Kubemark. And this is exactly one example.
And in my opinion such things should be handle directly in measurement - if measurement doesn't make sense in Kubemark, it should just print some warning or sth like that and not start. This will solve the problem of kubemark.

Going to the problem of disabling (or gradually enabling) stuff - I don't think it's the right approach. We should try to keep all the tests consistent. There is a question how we should enable stuff, but how I see that is that we may create a temporary fork of config and have an "experimental" job that will be using that config. And once we believe it works, we will be graduating it to the main config (for all setups).

Thoughts?

Reply:
Sounds good. Added a code that disables probes in kubemark and will enable the probes gradually via templates. 
Thanks!